src/coreclr/gc/gc.cpp

```diff
@@ -7120,36 +7122,48 @@ void gc_heap::fix_allocation_context (alloc_context* acontext, BOOL for_gc_p,
                  (size_t)acontext,
                  (size_t)acontext->alloc_ptr, (size_t)acontext->alloc_limit));
 
+    if (acontext->alloc_ptr == 0)
+    {
+        return;
+    }
+
     int align_const = get_alignment_constant (TRUE);
 
-    if (((size_t)(alloc_allocated - acontext->alloc_limit) > Align (min_obj_size, align_const)) ||
+#ifdef USE_REGIONS
+    heap_segment* region = find_segment(acontext->alloc_limit, FALSE);
```

Comment:
I'm guessing the reason why you are calling `find_segment` instead of `in_range_for_segment` as we talked about was you wanted to include UOH but this address will never be on UOH because it's a thread alloc context and we don't ever use thread ACs for UOH allocations. so I would still suggest to use in_range_for_segment, however I do need to make an additional suggestion which is to 

`bool is_ephemeral_heap_segment = in_range_for_segment ((acontext->alloc_limit - 1), ephemeral_heap_segment);`

because acontext->alloc_limit can be the same as heap_segment_reserved (ephemeral_heap_segment).

Reply:
Actually, I need the `region` for line 7135 as well in case the `alloc_limit - 1` is on a region other than the `ephemeral_heap_segment`.

Good catch for -1, I guess I would use `alloc_ptr` instead.

Reply:
> Actually, I need the region for line 7135 as well in case the alloc_limit - 1 is on a region other than the ephemeral_heap_segment.

ahh right, of course. makes sense. 

> I guess I would use alloc_ptr instead.

you'd need to do the same check for alloc_ptr - it could be the same as alloc_limit.

Reply:
so I see this is marked as resolved but appears it still uses alloc_ptr instead of (alloc_ptr - 1)?

Reply:
This is active now

Reply:
Fixed